internal/commands/build_test.go

```diff
@@ -931,6 +930,56 @@ builder = "my-builder"
 			})
 		})
 
+		when("path to app dir or zip-formatted file is provided", func() {
+			it("builds with the specified path", func() {
+				mockClient.EXPECT().
+					Build(gomock.Any(), EqBuildOptionsWithPath("my-source")).
+					Return(nil)
+
+				command.SetArgs([]string{"image", "--builder", "my-builder", "--path", "my-source"})
+				h.AssertNil(t, command.Execute())
+			})
+		})
+
+		when("a local path with the same string as the specified image name exists", func() {
+			var dir string
+
+			it.Before(func() {
+				// avoid using paths generated by os.MkdirTemp() as they cause test failures on macOS.
+				dir = "my-app-dir" + h.RandString(8)
+				err := os.Mkdir(dir, 0700)
```

Comment:
A path created by os.MkdirTemp() on macOS is like: `/var/folders/vh/0l94sp8n6hx8qwhb6wsy_xrh0000gr/T/tmp.4ffiJJd5rq`
It triggers an error in the validation of the image name when specified as a positional argument (image name).

Reply:
Are we creating the `my-app-dir` folder in the current directory? if for some reason the test fails and I am running the tests locally, will I see this new `my-app-dir` like a new file not being tracked by git?

Reply:
I agree that creating a new directory in the local source is a bit weird. An alternative would be to invoke `pack build testdata` since (I'm pretty sure) this test runs in the context of <pack source root>/internal/commands/. We could also try some combination of `os.MkdirTemp` and `os.Chdir` to create a directory with a sensible reference, but that might be a bit heavy-handed for what we need.

Reply:
@jjbustamante @natalieparellano 
Thank you for your comments and suggestions! I'll try to fix it to use the `testdata`. 🙏

Reply:
Done: https://github.com/buildpacks/pack/pull/2256/commits/3aee96ea4f82385680d76108099c24b393182fea

Now the code looks clean, and there will be no residue left in case of test failures. 😌